server/datastore/mysql/apple_mdm.go

```diff
@@ -543,8 +551,7 @@ func upsertMDMAppleHostLabelMembershipDB(ctx context.Context, tx sqlx.ExtContext
 func (ds *Datastore) UpdateHostTablesOnMDMUnenroll(ctx context.Context, uuid string) error {
 	return ds.withTx(ctx, func(tx sqlx.ExtContext) error {
 		_, err := tx.ExecContext(ctx, `
-			UPDATE host_mdm
-			SET enrolled = 0
+			DELETE FROM host_mdm
```

Comment:
Weirdly enough, it looks like we don't have a unique index on `uuid`? The only index it belongs in is a fulltext one, which I don't think can be used for strict equality as is the case here. I don't know if we had a good reason not to have a unique key, but for performance we might want to add at least a non-unique index on that column to avoid a table scan on a big table (hosts) (and use `LIMIT 1` in the select in the where)? Maybe this is premature and we should run some load tests with unenrollments, but this is possibly a concern at scale (80-100K+ hosts).

Reply:
> almost all MDM operations rely on matching the host by UUID, so this will be a general concern, not only on unenrollment

ah yeah, then I think it's definitely worth having an index.

> I'm tempted to send a sibling PR that adds the index and load test if it affects us negatively when doing INSERTs in hosts

Yeah that sounds good to me, this way we can land this PR faster and follow-up when it's more convenient with the index/load test separately.